AxonDeepSeg/morphometrics/compute_morphometrics.py

```diff
@@ -164,11 +171,11 @@ def get_axon_morphometrics(im_axon, path_folder=None, im_myelin=None, pixel_size
                     stats['axonmyelin_perimeter'] = axonmyelin_perimeter
                 except ZeroDivisionError:
                     print(f"ZeroDivisionError caught on invalid object #{idx}.")
-                    stats['gratio'] = float('NaN')
-                    stats['myelin_thickness'] = float('NaN')
-                    stats['myelin_area'] = float('NaN')
-                    stats['axonmyelin_area'] = float('NaN')
-                    stats['axonmyelin_perimeter'] = float('NaN')
+                    stats['gratio'] = 0
+                    stats['myelin_thickness'] = 0
+                    stats['myelin_area'] = 0
+                    stats['axonmyelin_area'] = 0
+                    stats['axonmyelin_perimeter'] = 0
```

Comment:
Hi @mariehbourget  - this is something we want to bring up again with the broader team.  I encountered an issue with the old syntax (NaNs) and could not resolve it any other way than by setting them to a numerical value. Essentially, some of my tests (I think in particular, checking that a saved and then loaded stats file is the same as the original variable) was failing in the presence of NaNs, which we were now encountering with one of the testing images now with your new IVADOMED model. I looked into it, and found that the assert equal evaluations don't work well for numpy arrays containing NaNs. I tried a few "tricks" but was unsuccessful, so decided on just setting it to 0 for now. By just switching the NaNs to zeroes, the test passed.

We could revert back to NaNs, but it will need a lot more efforts on how to "assert" these, which I think may just be a little bit of a waste of time imo. Should we bring this up to our slack channel?

Reply:
See the second test failure here: https://github.com/neuropoly/axondeepseg/runs/4490334934?check_suite_focus=true

Reply:
Grosso modo, it's because of this behaviour of NaNs:
<img width="501" alt="Screen Shot 2021-12-17 at 11 46 24 PM" src="https://user-images.githubusercontent.com/1421029/146627887-06201416-f1f8-4683-96bb-67130a516635.png">

Reply:
Oh! I see, thanks for the detailed explanation.

Would there be a way to test the following:
```
if a is NaN:
    assert that b is NaN as well
else:
    assert a==b
```

Also, what is the type of the NaN value once reloaded from the csv file, it may be a "nan" string and not a "number" making the `np.array_equal` fails in `test_load_axon_morphometrics_returns_identical_var_as_was_saved`?

Reply:
I don't think this would be an appropriate workaround, as we're dealing with arrays (or numpy arrays, I don't recall). 

Why was it important to have the values set as NaN instead of numerical values in the excel/csv table? Having all values numerical would be more consistent across formats (e.g. python value, csv, excel, etc), and reloading saved values would thus be consistent (otherwise, saving a NaN to csv might create a blank value, and then re-loading it may not produce a NaN again in Python, which is a problem I encountered or something similar at least).

Reply:
@mariehbourget I think I've resolved it. I changed it from `0` to `np.nan`, and the tests all pass (surprisingly). Looking back, it may have been the old implementation that was incorrect, i.e. the use of `float('NaN')` instead of numpy's built-in NaN..

Reply:
Nevermind, I had forgotten to change the default values from 0 to NaN. The error persists, and I'm having trouble asserting that two identical arrays are the same if they contain NaN and/or converting all NaN to 0 in it (I tried using numpy and pandas), because we have more of an array of dictionnaries rather than just numerical arrays.

Reply:
Ok so this time I've resolved it.

I had to write up a non-elegant way of testing the stats file, basically doing axon by axon and skipping those that have at least one NaN in the morphometrics dict. See below:

https://github.com/neuropoly/axondeepseg/blob/ce7c4dba35495886cf292245ba19ad3b01d04c46/test/morphometrics/test_compute_morphometrics.py#L650-L654